concepts/websockets.textile

```diff
@@ -0,0 +1,593 @@
+
+As per the "deepstream server website":https://deepstreamhub.com/open-source/, _"deepstream is a powerful websocket server that syncs realtime data between browsers, smartphones, backends and the IoT"_. Unlike the other server-based libraries above, deepstream server is its own server application which must be downloaded and installed on your server. Bindings are provided for "integration with different languages and environments":https://deepstreamhub.com/docs/, such as Node.js and Java. As a full-featured server, it comes with a full suite of capabilities, including pub/sub channels, presence, authentication, and more.
+
+h2(#scaling-beyond-a-single-server). Scaling beyond a single server
+
+The number of concurrent connections a server can handle is rarely the bottleneck when it comes to server load. Most decent WebSocket servers can support hundreds of thousands or even millions of concurrent connections, but what's the workload required to process and respond to messages once the WebSocket server process has handled receipt of the actual data? Typically there will be all kinds of potential concerns, such as reading and writing to and from a database, integration with a game server, allocation and management of resources for each client, and so forth. As soon as one machine is unable to cope with the workload, you'll need to start adding additional servers, which means now you'll need to start thinking about load-balancing, synchronization of messages among clients connected to different servers, generalised access to client state irrespective of connection lifespan or the specific server that the client is connected to - the list goes on and on.
+
+Such concerns are deserving of an article all of their own, and you'll find plenty in "the _Engineering_ section of the Ably blog":https://blog.ably.io - see in particular Matt O'Riordan's writeup on "some of the complexities that were dealt with when building Ably":https://blog.ably.io/the-story-of-how-ably-came-to-be-1d859f2c9a90.
+
+h2(#ably-realtime-messaging-as-a-service). Ably: Realtime messaging as a service
+
+As you have seen, there's a lot involved when implementing support for The WebSocket protocol, not just in terms of client and server implementation details, but also with respect to support for other transports to ensure robust support for different client environments, as well as larger concerns, such as authentication and authorization, guaranteed message delivery, reliable message ordering, historical message retention, and so forth.
+
+Ably is a battle-tested, enterprise-ready, cloud-based messaging platform with native support for WebSockets, long-polling and other relevant transports and protocols. We've already done all of the above work for you, which means you can get on with the real task of building your business's application platform and leave the mammoth task of implementing robust, reliable realtime messaging to us.
+
+"Find out more here":https://ably.io.
```

Comment:
I hate text which says here, it's unnecessary and semantically bad.  How about "Find out more about Ably"